packages/react/dialog/src/Dialog.tsx

```diff
@@ -12,6 +12,7 @@ import { Primitive } from '@radix-ui/react-primitive';
 import { useFocusGuards } from '@radix-ui/react-focus-guards';
 import { RemoveScroll } from 'react-remove-scroll';
 import { hideOthers } from 'aria-hidden';
+import { Slot } from '@radix-ui/react-slot';
```

Comment:
I realised that the `forwardProps` solution on `RemoveScroll` was breaking unmount animation... no idea why. Our `Slot` seems to combine the two better and things all work.

Reply:
Cool, and that's what we used to do before right? `as={Slot}` because they also have an `as`.

Reply:
Yep 👍